ReflectometryServer/components.py

```diff
@@ -1,45 +1,32 @@
 """
 Components on a beam
 """
+from ReflectometryServer.beam_path_calc import TrackingBeamPathCalc, BeamPathTilting, BeamPathCalcAngleReflecting, \
+    BeamPathCalcTheta
+from ReflectometryServer.movement_strategy import LinearMovementCalc
+import logging
 
-from ReflectometryServer.gemoetry import PositionAndAngle
+logger = logging.getLogger(__name__)
 
 
 class Component(object):
     """
     Base object for all components that can sit on a beam line
     """
 
-    def __init__(self, name, movement_strategy):
+    def __init__(self, name, setup):
         """
         Initializer.
         Args:
             name (str): name of the component
-            movement_strategy (VerticalMovement): strategy for calculating the interception between the movement of the
-            component and the incoming beam
+            setup (ReflectometryServer.movement_strategy.LinearSetup): initial setup for the component
         """
-        self.incoming_beam = None
-        self._movement_strategy = movement_strategy
-        self.after_beam_path_update_listener = lambda x: None
-        self._enabled = True
         self._name = name
+        self._init_beam_path_calcs(setup)
 
-    @property
-    def enabled(self):
-        """
-        Returns: the enabled status
-        """
-        return self._enabled
-
-    @enabled.setter
-    def enabled(self, enabled):
-        """
-        Updates the component enabled status and notifies the beam path update listener
-        Args:
-            enabled: The modified enabled status
-        """
-        self._enabled = enabled
-        self.after_beam_path_update_listener(self)
+    def _init_beam_path_calcs(self, setup):
+        self._beam_path_set_point = TrackingBeamPathCalc(LinearMovementCalc(setup))
+        self._beam_path_rbv = TrackingBeamPathCalc(LinearMovementCalc(setup))
```

Comment:
Yes I think so. It can be undone if needed but there was too much being passed down from the top. 

Component("s1", PositionAndAngle(0.0, 1, perp_to_floor))

becomes

Component("s1", TrackingBeamPathCalc(LinearMovementCalc(PositionAndAngle(0.0, 1, perp_to_floor)), TrackingBeamPathCalc(LinearMovementCalc(PositionAndAngle(0.0, 1, perp_to_floor)))

Although possibly what we really need is a builder at the config level. Maybe we should think about that.